### PR TITLE
DRAFT Updated the pump device

### PR DESCRIPTION
**Pull request by TaylorSmith28**

### CODE REVIEW

File: titration/ui_state/demo_mode/demo_pump.py [hunk 1]
```diff
@@ -16,43 +18,68 @@ def handle_key(self, key):
...
         The function to handle keypad input:
             Substate 1:
-                1 -> Get Volume
-                2 -> Set Volume
-                3 -> Pull Volume in to Pump
-                4 -> Go to 2nd page of options
+                1 -> Get Pump Volume
+                2 -> Get Added Volume
+                3 -> Clear Added Volume
+                4 -> Got 2nd page of options
```
Comment (prestoncarman) [inline titration/ui_state/demo_mode/demo_pump.py:24]: Go to...

File: titration/ui_state/demo_mode/demo_pump.py [hunk 2]
```diff
@@ -16,43 +18,68 @@ def handle_key(self, key):
...
+                2 -> Empty Pump
+                3 -> Pump In
+                4 -> Go to 3rd page of options
             Substate 3:
-                Any -> Substate 1
+                1 -> Pump Out
+                4 -> Go to 1st page
             Substate 4:
                 Any -> Substate 1
+            Substate 5:
```
Comment (prestoncarman) [inline titration/ui_state/demo_mode/demo_pump.py:35]: Why substate 4 and 5?

File: titration/ui_state/user_value/pump_volume.py [hunk 3]
```diff
@@ -13,7 +13,8 @@ def save_value(self, value):
         """
         The function to save the titrator's pump volume
         """
-        self.titrator.pump_volume = value
+        if value not in ("", "."):
```
Comment (prestoncarman) [inline titration/ui_state/user_value/pump_volume.py:16]: What does this statement do?

File: test/ui_state/demo_mode/demo_pump_test.py [hunk 4]
```diff
@@ -136,28 +289,70 @@ def test_demo_mode(print_mock, set_next_state_mock):
...
+    )
 
     demo_pump.handle_key("3")
+    assert demo_pump.substate == 6
+
+    demo_pump.loop()
+    print.assert_has_calls(
+        [
+            mock.call("Cleared", line=1, style="center"),
+            mock.call(
```
Comment (prestoncarman) [inline test/ui_state/demo_mode/demo_pump_test.py:338]: Why is this on multiple lines?

File: test/ui_state/demo_mode/demo_pump_test.py [hunk 5]
```diff
@@ -166,22 +361,146 @@ def test_demo_mode(print_mock, set_next_state_mock):
...
+
+    demo_pump.handle_key("1")
+    assert demo_pump.substate == 7
+    fill.assert_called()
+
+    demo_pump.loop()
+    print.assert_has_calls(
+        [
+            mock.call("Filling Pump", line=1),
+            mock.call(
```
Comment (prestoncarman) [inline test/ui_state/demo_mode/demo_pump_test.py:381]: Why on multiple lines?

**Review by prestoncarman — CHANGES_REQUESTED** [review]

File: titration/ui_state/demo_mode/demo_pump.py (hunk 1 above)
Comment (TaylorSmith28) [reply titration/ui_state/demo_mode/demo_pump.py]: Fixed

File: titration/ui_state/demo_mode/demo_pump.py (hunk 2 above)
Comment (TaylorSmith28) [reply titration/ui_state/demo_mode/demo_pump.py]: Fixed

File: titration/ui_state/user_value/pump_volume.py (hunk 3 above)
Comment (TaylorSmith28) [reply titration/ui_state/user_value/pump_volume.py:16]: It ensures that the value entered in the user_value states is an actual number.

File: test/ui_state/demo_mode/demo_pump_test.py (hunk 4 above)
Comment (TaylorSmith28) [reply test/ui_state/demo_mode/demo_pump_test.py:338]: It pleases the Linters

File: titration/ui_state/user_value/pump_volume.py (hunk 3 above)
Comment (TaylorSmith28) [reply titration/ui_state/user_value/pump_volume.py]: This is fixed in #115

**Pull request closed without merging**